### PR TITLE
feat(ui): tap video duration to toggle remaining time counter

### PR DESCRIPTION
**Pull request by IndusAryan**
@fire-light42 shifted to views

### CODE REVIEW

**Comment by fire-light42** [comment]
Why is this draft? do you want a review?

**Comment by IndusAryan** [comment]
@fire-light42 i thought you will ask some minor changes, anyway it can be merged, undrafting it now

**Comment by fire-light42** [comment]
The code looked fine, **HOWEVER** when testing it I found that it was unsynced af making it feel shitty and it does not save the setting between runs. 

The first issue can be solved by binding the update to the text change of the position set by exo, this also removes the need to a looper
```kotlin
exoPosition.doOnTextChanged { _, _, _, _ ->
    updateRemainingTime()
}
``` 
and dont forget
```kotlin
val remainingTimeSeconds = (duration - position + 500) / 1000 // this is due to rounding
```
The second issue can be solved with a PreferenceDelegate
eg `var resizeMode : Int by UserPreferenceDelegate("resize_mode", 0)` with a bool instead

**Review by fire-light42 — CHANGES_REQUESTED** [review]
See above

**Comment by IndusAryan** [comment]
> See above

so i made and tested these desired changes ,  now it is going with sync properly and i also replaces the comment // cs is ott itself to cs3 is media center itself so that it dont provoke  any  unwanted heat or issue and Thanks for telling about preference delegates, i had no idea about them or their existence .

can you tell me about this method what does _ _ _ do, do they ignore parameters that are not present
```  playerBinding?.exoPosition?.doOnTextChanged { _, _, _, _ ->
            updateRemainingTime()
        } ```
        
        because i experimented first with
        ```  playerBinding?.exoPosition?.doOnTextChanged (
            updateRemainingTime())
             ```
             but it crashed and turned update function to 
             ```char sequence , int , int  -> unit ......
             return updatingtime // this was flagged as recursive call by ide```
             
             finally there is one thing is that we call stopupdating remainingtime onDestroy , and that turn preference to 0 , so the setting only applies to a session, for example if user goes to next episode his seting work, but in different video file he has to press duration again , this can be repaired by not calling the stop on destroy but idk what repurcussion it can have in both tv or mobile.

**Comment by IndusAryan** [comment]
because i experimented first with,
        
        ```kotlin
        playerBinding?.exoPosition?.doOnTextChanged (
            updateRemainingTime())```
            
             but it crashed and turned update function to 
             ```char sequence , int , int  -> unit ......
             return updatingtime // this was flagged as recursive call by ide```
             
             finally there is one thing is that we call stopupdating remainingtime onDestroy , and that turn preference to 0 , so the setting only applies to a session, for example if user goes to next episode his seting work, but in different video file he has to press duration again , this can be repaired by not calling the stop on destroy but idk what repurcussion it can have in both tv or mobile.

**Review by fire-light42 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/lagradost/cloudstream3/ui/player/FullScreenPlayer.kt [hunk 1]
```diff
@@ -1559,6 +1540,36 @@ open class FullScreenPlayer : AbstractPlayerFragment() {
...
+            val remainingTimeSeconds = (duration - position + 500) / 1000
+            val formattedTime = "-${DateUtils.formatElapsedTime(remainingTimeSeconds)}"
+
+            playerBinding?.timeLeft?.text = formattedTime
+        }
+    }
+
+    private fun startUpdatingRemainingTime() {
+
+        playerBinding?.exoPosition?.doOnTextChanged { _, _, _, _ ->
```
[the remark this answers is not in the document]
Comment (IndusAryan) [reply app/src/main/java/com/lagradost/cloudstream3/ui/player/FullScreenPlayer.kt]: i think i got it thsi time

File: app/src/main/java/com/lagradost/cloudstream3/ui/player/FullScreenPlayer.kt [hunk 2]
```diff
@@ -1559,6 +1540,36 @@ open class FullScreenPlayer : AbstractPlayerFragment() {
...
+
+        if (duration != null && duration > 1 && position != null) {
+            val remainingTimeSeconds = (duration - position + 500) / 1000
+            val formattedTime = "-${DateUtils.formatElapsedTime(remainingTimeSeconds)}"
+
+            playerBinding?.timeLeft?.text = formattedTime
+        }
+    }
+
+    private fun startUpdatingRemainingTime() {
```
[the remark this answers is not in the document]
Comment (IndusAryan) [reply app/src/main/java/com/lagradost/cloudstream3/ui/player/FullScreenPlayer.kt]: created

File: app/src/main/java/com/lagradost/cloudstream3/ui/player/FullScreenPlayer.kt [hunk 3]
```diff
@@ -79,10 +78,9 @@ open class FullScreenPlayer : AbstractPlayerFragment() {
     protected open var lockRotation = true
     protected open var isFullScreenPlayer = true
     protected open var isTv = false
-
     protected var playerBinding: PlayerCustomLayoutBinding? = null
 
-
+    private var durationMode : Int by UserPreferenceDelegate("duration_mode", 0)
```
[the remark this answers is not in the document]
Comment (IndusAryan) [reply app/src/main/java/com/lagradost/cloudstream3/ui/player/FullScreenPlayer.kt]: yes

**Review by fire-light42 — APPROVED** [review]

**Pull request merged**